xen/arch/arm/coproc/coproc.c

```diff
@@ -353,24 +350,29 @@ struct coproc_device *coproc_alloc(struct platform_device *pdev,
 
     for ( i = 0; i < num_mmios; ++i )
     {
-        res = platform_get_resource(pdev, IORESOURCE_MEM, i);
-        coproc->mmios[i].base = devm_ioremap_resource(dev, res);
+        ret = dt_device_get_address(pdev, i,
+                                    &coproc->mmios[i].addr,
+                                    &coproc->mmios[i].size);
```

Comment:
I would add next:
struct mmio *mmio = &coproc->mmios[i];

And then use mmio->xxx instead of coproc->mmios[i].xxx constuction